processing/src/main/java/io/druid/query/aggregation/post/ArithmeticPostAggregator.java

```diff
@@ -47,25 +48,40 @@ public int compare(Object o, Object o1)
   private final String fnName;
   private final List<PostAggregator> fields;
   private final Ops op;
+  private final Comparator comparator;
+  private final String ordering;
+
+  public ArithmeticPostAggregator(
+      String name,
+      String fnName,
+      List<PostAggregator> fields
+  )
+  {
+    this(name, fnName, fields, null);
+  }
 
   @JsonCreator
   public ArithmeticPostAggregator(
       @JsonProperty("name") String name,
       @JsonProperty("fn") String fnName,
-      @JsonProperty("fields") List<PostAggregator> fields
+      @JsonProperty("fields") List<PostAggregator> fields,
+      @JsonProperty("ordering") String ordering
   )
   {
+    Preconditions.checkArgument(fnName != null, "fn cannot not be null");
+    Preconditions.checkArgument(fields != null && fields.size() > 1, "Illegal number of fields[%s], must be > 1");
+
     this.name = name;
     this.fnName = fnName;
     this.fields = fields;
-    if (fields.size() <= 1) {
-      throw new IAE("Illegal number of fields[%s], must be > 1", fields.size());
-    }
 
     this.op = Ops.lookup(fnName);
     if (op == null) {
       throw new IAE("Unknown operation[%s], known operations[%s]", fnName, Ops.getFns());
     }
+
+    this.ordering = ordering;
+    this.comparator = ordering == null ? DEFAULT_COMPARATOR : Ordering.valueOf(ordering);
```

Comment:
I think natural ordering is a very reasonable default to expect, regardless of query type and operator.

Changing the ordering for specific operators may get very confusing, for instance `sum(a) / sum(b)` and `(sum(a) / sum(b) * 100)` would suddenly have a different ordering because `*` and `quotient` have a different default comparator.